src/full/Agda/TypeChecking/Monad/Modality.hs

```diff
@@ -0,0 +1,200 @@
+{- | Modality.
+
+Agda has support for several modalities, namely:
+
+ * 'Cohesion'
+ * 'Quantity'
+ * 'Relevance'
+
+In order to type check such modalities, we must store the current modality in
+the typing context. This module provides functions to update the context based
+on a given modality.
+
+See 'Agda.TypeChecking.Irrelevance'.
```

Comment:
Module references need to be in double quotes in haddock, if I remember correctly.

Reply:
You're entirely right, thank you! It generates a link to the `Irrelevance` section of the (non-existent) `Agda.TypeChecking` module.